pkg/webhook/cspc_test.go

```diff
@@ -1128,6 +1228,72 @@ func TestCSPCScaleDown(t *testing.T) {
 			expectedRsp: true,
 			getCSPCObj:  getCSPCObject,
 		},
+		"Scale down when node doesn't exist in cluster": {
+			existingObj: cstor.NewCStorPoolCluster().
+				WithName("cspc-bar-mirror-2").
+				WithNamespace("openebs").
+				WithPoolSpecs(
+					*cstor.NewPoolSpec().
+						WithNodeSelector(map[string]string{types.HostNameLabelKey: "worker-not-exist"}).
+						WithPoolConfig(*cstor.NewPoolConfig().
+							WithDataRaidGroupType("mirror")).
+						WithDataRaidGroups(*cstor.NewRaidGroup().
+							WithCStorPoolInstanceBlockDevices(
+								*cstor.NewCStorPoolInstanceBlockDevice().WithName("blockdevice-5"),
+								*cstor.NewCStorPoolInstanceBlockDevice().WithName("blockdevice-6"),
+							),
+						),
+					*cstor.NewPoolSpec().
+						WithNodeSelector(map[string]string{types.HostNameLabelKey: "worker-2"}).
+						WithPoolConfig(*cstor.NewPoolConfig().
+							WithDataRaidGroupType("mirror")).
+						WithDataRaidGroups(*cstor.NewRaidGroup().
+							WithCStorPoolInstanceBlockDevices(
+								*cstor.NewCStorPoolInstanceBlockDevice().WithName("blockdevice-27"),
+								*cstor.NewCStorPoolInstanceBlockDevice().WithName("blockdevice-28"),
+							),
+						),
+				),
+			requestedObj: cstor.NewCStorPoolCluster().
+				WithName("cspc-bar-mirror-2").
+				WithNamespace("openebs").
+				WithPoolSpecs(
+					*cstor.NewPoolSpec().
+						WithNodeSelector(map[string]string{types.HostNameLabelKey: "worker-2"}).
+						WithPoolConfig(*cstor.NewPoolConfig().
+							WithDataRaidGroupType("mirror")).
+						WithDataRaidGroups(*cstor.NewRaidGroup().
+							WithCStorPoolInstanceBlockDevices(
+								*cstor.NewCStorPoolInstanceBlockDevice().WithName("blockdevice-27"),
+								*cstor.NewCStorPoolInstanceBlockDevice().WithName("blockdevice-28"),
+							),
+						),
+				),
+			existingCSPIs: []*cstor.CStorPoolInstance{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cspc-bar-mirror-2-cspi-1",
+						Namespace: "openebs",
+						Labels: map[string]string{
+							types.HostNameLabelKey:         "worker-not-exist",
```

Comment:
How does this simulate -- node does not exist behavior?

Reply:
First, we persisted CSPC in etcd with a pool spec belongs to a node that doesn't exist in etcd. During CSPC update time we removed the pool spec belongs to the node that it doesn't exist in the cluster.

At very high-level this test scenario is like:
- CSPC provisioned the pools on node1, node2, and node3 it interns creates CSPIs on specified nodes.
- Now let us assume one of the nodes lets say node2 has been removed from the cluster and user-triggered scaledown by removing pool spec belongs to node2 now admission server shouldn't reject the request saying not able to fetch the node using node selector(Without this PR this test case will fail).

Reply:
Moreover here it is component `UnitTest` case.